### ISSUE TITLE
[a11y]: The <label> element for checkbox throws error for table with checboxes

### ISSUE THREAD

**Issue by r1shabhsharma** [carbon-design-system/carbon#20017]
### Package

@carbon/web-components

### Browser

Chrome

### Package version

2.27.2

### Automated testing tool and ruleset

IBM Equal Access Accessibility Checker

### Description

The error msg `The <label> element does not have descriptive text that identifies the expected input` is shown for the checkboxes of table header and rows 

### Reproduction/example

Simply try running test on web components table with checkboxes

### Steps to reproduce

Just run the accessibility tool on the table with checkboxes. The issue will pop up

### Suggested Severity

None

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems
